src/Library/demos/Text Input/main.blp

```diff
@@ -0,0 +1,309 @@
+      }
+    }
+
+    Separator {
+      margin-bottom: 20;
+    }
+
+    Label {
+      label: "Search Entry";
+      halign: start;
+
+      styles [
+        "heading",
+      ]
+    }
+
+    Label {
+      label: "An entry widget that has been tailored for searching.";
+      margin-top: 10;
+      margin-bottom: 10;
+      halign: start;
+
+      styles [
+        "dim-label",
+      ]
+    }
+
+    SearchEntry search_entry {
+      margin-bottom: 10;
+      placeholder-text: "Search for items...";
+    }
+
+    ListBox items {
+      margin-bottom: 10;
+
+      styles [
+        "boxed-list",
+      ]
+
+      ListBoxRow {
+        Label {
+          label: "Apples";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Oranges";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Bananas";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Strawberries";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Pineapples";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+
+      ListBoxRow {
+        Label {
+          label: "Blueberries";
+          margin-top: 10;
+          margin-bottom: 10;
+        }
+      }
+    }
+
+    LinkButton {
+      label: "API Reference";
+      uri: "https://docs.gtk.org/gtk4/class.SearchEntry.html";
+      margin-top: 20;
+      margin-bottom: 20;
+    }
+
+    Separator {
+      margin-bottom: 20;
+    }
+
+    Label {
+      label: "Password Entry";
+      halign: start;
+
+      styles [
+        "heading",
+      ]
+    }
+
+    Label {
+      label: "An entry with dedicated control for entering secrets and passwords.";
+      margin-top: 10;
+      margin-bottom: 10;
+      margin-end: 10;
+      halign: start;
+
+      styles [
+        "dim-label",
+      ]
+    }
+
+    PasswordEntry passwd_entry{
+      show-peek-icon: true;
+      placeholder-text: "Enter password";
+      margin-bottom: 10;
+    }
+
+    PasswordEntry confirm_passwd_entry{
+      show-peek-icon: true;
+      placeholder-text: "Confirm password";
+      margin-bottom: 10;
+    }
+
+    Label passwd_label {
+    }
+
+    Box {
+      halign: center;
+      margin-top: 5;
+      margin-bottom: 20;
+
+      LinkButton{
+        label: "Tutorial";
+        uri: "https://developer.gnome.org/documentation/tutorials/beginners/components/password_entry.html";
+      }
+
+      LinkButton{
+        label: "API Reference";
+        uri: "https://docs.gtk.org/gtk4/class.PasswordEntry.html";
+      }
+    }
+
+    LinkButton{
+      label: "Human Interface Guidelines";
+      uri: "https://developer.gnome.org/hig/patterns/controls/text-fields.html";
+      margin-bottom:5;
+    }
```

Comment:
maybe we move this one above the "Entry" section so it doesn't look like it's for password

Reply:
👍🏾